### PR TITLE
Add zettcache device support, DWARF call frame, and more perf stats collection

### PR DESCRIPTION
**Pull request by tonynguien**
### Motivation and Context

DOSE-564 ZTS - Enable DWARF call frame for perf-record
DOSE-559 ZTS - Add network throughput and arcstat to data collected in tests
DOSE-573 ZTS - Add zettacache support

### How Has This Been Tested?

Ran ZFS performance test suite and verify dstat, arcstat, and perf files are generated. And `zfs_object_agent` runs with specified zettacache device.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: scripts/zfs-tests.sh [hunk 1]
```diff
@@ -610,8 +610,14 @@ if [ -n "$ZTS_OBJECT_STORE" ]; then
 	# Start zfs_object_agent service and redirect the output to ZOA_LOG
 	# file.
 	#
-	sudo -E /sbin/zfs_object_agent -vv \
-	    --output-file=$ZOA_LOG >/dev/null 2>&1 &
+	if [ -n "$ZETTA_CACHE_DEV" ]; then
+		dev=$(basename "$ZETTA_CACHE_DEV")
```
Comment (nupur-agrawal-delphix) [inline scripts/zfs-tests.sh:614]: Here, we are assuming that we will always have just 1 zetta cache device. What if we have multiple?

File: tests/zfs-tests/tests/perf/regression/random_reads.ksh [hunk 2]
```diff
@@ -90,11 +90,21 @@ export FILE_SIZE=$((TOTAL_SIZE / NUMJOBS))
 export DIRECTORY=$(get_directory)
 log_must fio $FIO_SCRIPTS/mkfiles.fio
 
+if [ -n "$ZETTA_CACHE_DEV" ]; then
```
Comment (jwk404) [inline tests/zfs-tests/tests/perf/regression/random_reads.ksh:93]: What would you think about making this a function in perf.shlib?

File: scripts/zfs-tests.sh [hunk 3]
```diff
@@ -610,8 +610,14 @@ if [ -n "$ZTS_OBJECT_STORE" ]; then
 	# Start zfs_object_agent service and redirect the output to ZOA_LOG
 	# file.
 	#
-	sudo -E /sbin/zfs_object_agent -vv \
-	    --output-file=$ZOA_LOG >/dev/null 2>&1 &
+	if [ -n "$ZETTA_CACHE_DEV" ]; then
```
Comment (jwk404) [inline scripts/zfs-tests.sh:613]: Can we add a line like the others at 699 that prints the value of this variable when the run starts?

**Review by jwk404 — CHANGES_REQUESTED** [review]

File: scripts/zfs-tests.sh (hunk 1 above)
Comment (tonynguien) [reply scripts/zfs-tests.sh:614]: Zettacache code supports only a single device at the moment. IMO, we can update this logic when we have multiple devices support.

File: tests/zfs-tests/tests/perf/regression/random_reads.ksh (hunk 2 above)
Comment (tonynguien) [reply tests/zfs-tests/tests/perf/regression/random_reads.ksh:93]: Good idea! I'll do that.

File: scripts/zfs-tests.sh (hunk 1 above)
Comment (nupur-agrawal-delphix) [reply scripts/zfs-tests.sh:614]: Ah I didn't know that. Thanks !

File: scripts/zfs-tests.sh (hunk 3 above)
Comment (tonynguien) [reply scripts/zfs-tests.sh:613]: will do.

**Review by jwk404 — APPROVED** [review]

**Pull request merged**